buildSrc/src/main/groovy/nva-publication-api.java-common-conventions.gradle

```diff
@@ -0,0 +1,156 @@
+    environment "API_HOST", "localhost"
+    environment "API_SCHEME", "https"
+}
+
+
+jacocoTestReport {
+    reports {
+        xml.enabled true
+    }
+}
+
+//gradleLint {
+//    rules = ['unused-dependency']
+//}
+
+pmd {
+    ruleSetConfig = rootProject.resources.text.fromFile('config/pmd/ruleset.xml')
+    ruleSets = []
+    ignoreFailures = false
+}
+
+checkstyle {
+    configFile = rootProject.resources.text.fromFile('config/checkstyle/checkstyle.xml').asFile()
+    showViolations = true
+}
+
+tasks.withType(Checkstyle) {
+    reports {
+        xml.enabled false
+        html.enabled true
+        html.stylesheet rootProject.resources.text.fromFile('config/checkstyle/checkstyle-simple.xsl')
+    }
+}
+
+check.dependsOn jacocoTestCoverageVerification
+jacocoTestCoverageVerification.dependsOn(jacocoTestReport)
+
+jacocoTestCoverageVerification {
+    violationRules {
+        rule {
+            limit {
+                counter = 'METHOD'
+                value = 'COVEREDRATIO'
+                minimum = 1.000
+            }
+        }
+
+        rule {
+            limit {
+                counter = 'CLASS'
+                value = 'COVEREDRATIO'
+                minimum = 1.000
+            }
+        }
+    }
+}
+
+// publish submodules
+task sourcesJar(type: Jar, dependsOn: classes) {
+    archiveClassifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+artifacts {
+    archives sourcesJar
+}
+
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
+            artifactId = project.name
+            from components.java
+            versionMapping {
+                usage('java-api') {
+                    fromResolutionOf('runtimeClasspath')
+                }
+                usage('java-runtime') {
+                    fromResolutionResult()
+                }
+            }
+            pom {
+                name = project.name
+                description = 'The nva publication service'
+                url = 'https://github.com/BIBSYSDEV/nva-publication-api'
+                licenses {
+                    license {
+                        name = 'MIT License'
+                        url = 'http://www.opensource.org/licenses/mit-license.php'
+                    }
+                }
+                developers {
+                    developer {
+                        name = 'Orestis Gkorgkas'
+                        email = 'og@unit.no'
+                        organization = 'UNIT'
+                        organizationUrl = 'https://www.unit.no/'
+                    }
+                }
+                scm {
+                    connection = 'scm:git:git://github.com/BIBSYSDEV/nva-publication-api.git'
+                    developerConnection = 'scm:git:ssh://github.com/BIBSYSDEV/nva-publication-api.git'
+                    url = 'https://github.com/BIBSYSDEV/nva-publication-api/tree/main'
+                }
+            }
+        }
+    }
+}
```

Comment:
Similarlly, not sure if this fits into a generalized plugin

Reply:
This feels like a more global plugin — some things will be globally true, others not